backend/fms_core/services/dataset.py

```diff
@@ -0,0 +1,195 @@
+from django.core.exceptions import ObjectDoesNotExist, ValidationError
+
+from datetime import datetime, date
+from typing import Any, Dict, List, Optional, Tuple, Union
+
+from fms_core.models.dataset_file import DatasetFile
+from fms_core.models.dataset import Dataset
+
+def convertToModelDate(value: Union[datetime, date, str, None]) -> Union[datetime, None]:
+    if isinstance(value, date):
+        return datetime.combine(value, datetime.min.time())
+    elif isinstance(value, datetime):
+        return value
+    elif isinstance(value, str):
+        return datetime.fromisoformat(value[:10])
+    elif value is None:
+        return value
+    else:
+        raise ValueError(f"Invalid date: {value}")
+
+def create_dataset(project_name: str, run_name: str, lane: str, files: List[Dict[str, Any]] = []) -> Tuple[Union[Dataset, None], List[str], List[str]]:
+    dataset = None
+
+    errors = []
+    warnings = []
+
+    try:
+        dataset = Dataset.objects.create(
```

Comment:
Try to anticipate the issues you could get from missing parameters and catch them before hitting the try as much as possible to be able to provide more useful error messages. This is unlikely to be propagated to the user in this case but we'll want to know what's wrong if it crashes.